### PR TITLE
Fix PBS linking against libcrypt.so.1 for real.

### LINKED ISSUE #52: Error while loading shared libraries: libcrypt.so.1
Known issue from Python Standalone (https://github.com/indygreg/python-build-standalone/issues/113) and PyOx (https://github.com/indygreg/PyOxidizer/issues/89).

Ran into this in Fedora (environment detailed below). Will try Greg's workaround tonight when I'm home.

```bash
⬢[sj@pants pants-plugins]$ ./scie-pants-linux-x86_64 
/var/home/sj/.cache/nce/cdc3a4cfddcd63b6cebdd75b14970e02d8ef0ac5be4d350e57ab5df56c19e85e/cpython-3.9.15+20221106-x86_64-unknown-linux-gnu-install_only.tar.gz/python/bin/python3.9: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
Error: Isolates your Pants from the elements.

Please select from the following boot commands:

<default>: Detects the current Pants installation and launches it.
bootstrap-tools: Introspection tools for the Pants bootstrap process.
pants: Runs a hermetic Pants installation.
pants-debug: Runs a hermetic Pants installation with a debug server for debugging Pants code.
update: Update scie-pants.

You can select a boot command by passing it as the 1st argument or else by setting the SCIE_BOOT environment variable.

ERROR: Failed to establish atomic directory /var/home/sj/.cache/nce/4babf95d09aea647ecb10481b524c5c2b4881e0338b465f2f2724356698955b6/locks/configure-48820e0e26193929b465ab5ca48693a932844b8082a9b529facbb28128a8ea85. Population of work directory failed: Boot binding command failed: exit status: 127
```

## Suggestions

Since we use Python 3.9 standalone, some number of Red Hat users will run into this bug when using Pants. I'd recommend documenting this problem and linking back to Greg's documentation (https://github.com/indygreg/python-build-standalone/blob/main/docs/quirks.rst#missing-libcryptso1).

Alternatively/In addition - would it be worth checking for that missing library, and providing a clearer and/or more actionable warning on Linux? 

## Environment

Fedora 37 toolbox created in Fedora SilverBlue 37 host OS.

```bash
⬢[sj@pants pants-plugins]$ neofetch 
             .',;::::;,'.                sj@pants 
         .';:cccccccccccc:;,.            -------- 
      .;cccccccccccccccccccccc;.         OS: Fedora Linux 37 (Container Image) x86_64 
    .:cccccccccccccccccccccccccc:.       Host: B550I AORUS PRO AX 
  .;ccccccccccccc;.:dddl:.;ccccccc;.     Kernel: 6.0.14-300.fc37.x86_64 
 .:ccccccccccccc;OWMKOOXMWd;ccccccc:.    Uptime: 3 days, 1 hour, 8 mins 
.:ccccccccccccc;KMMc;cc;xMMc:ccccccc:.   Packages: 475 (rpm) 
,cccccccccccccc;MMM.;cc;;WW::cccccccc,   Shell: bash 5.2.9 
:cccccccccccccc;MMM.;cccccccccccccccc:   Resolution: 2560x1080 
:ccccccc;oxOOOo;MMM0OOk.;cccccccccccc:   Terminal: conmon 
cccccc:0MMKxdd:;MMMkddc.;cccccccccccc;   CPU: AMD Ryzen 7 3800X (16) @ 3.900GHz 
ccccc:XM0';cccc;MMM.;cccccccccccccccc'   GPU: NVIDIA GeForce GTX 1660 SUPER 
ccccc;MMo;ccccc;MMW.;ccccccccccccccc;    Memory: 6503MiB / 31996MiB 
ccccc;0MNc.ccc.xMMd:ccccccccccccccc;
cccccc;dNMWXXXWM0::cccccccccccccc:,                              
cccccccc;.:odl:.;cccccccccccccc:,.                               
:cccccccccccccccccccccccccccc:'.
.:cccccccccccccccccccccc:;,..
  '::cccccccccccccc::;,.
```

### PR DESCRIPTION
**Pull request by jsirois**
The prior PBS upgrade was to a version that claimed to have this fixed,
but it was only fixed for 3.11 and not actually fixed for `3.{8,9,10}`.
This time we upgrade again to
https://github.com/indygreg/python-build-standalone/releases/tag/20240107
which notes this is actually now fixed for those Pythons. A manual test
of scie-pants 0.10.5 inside a fedora:37 container confirms the current
broken status and mounting in a freshly build scie-pants using this
change confirms a fix.

Fixes #52

### CODE REVIEW

**Comment by jsirois** [comment]
### Existing latest scie-pants:
```
$ docker run --rm -it fedora:37
[root@79358947d724 /]# curl --proto '=https' --tlsv1.2 -fsSL https://static.pantsbuild.org/setup/get-pants.sh | bash
Downloading and installing the pants launcher ...
Installed the pants launcher from https://github.com/pantsbuild/scie-pants/releases/latest/download/scie-pants-linux-x86_64 to /root/.local/bin/pants

Running `pants` in a Pants-enabled repo will use the version of Pants configured for that repo.
In a repo not yet Pants-enabled, it will prompt you to set up Pants for that repo.
[root@79358947d724 /]# /root/.local/bin/pants -V
/root/.cache/nce/f3ff38b1ccae7dcebd8bbf2e533c9a984fac881de0ffd1636fbb61842bd924de/cpython-3.9.18+20231002-x86_64-unknown-linux-gnu-install_only.tar.gz/python/bin/python3.9: error while loading shared libraries: libcrypt.so.1: cannot open shared object file: No such file or directory
Error: Failed to establish atomic directory /root/.cache/nce/3d6643e46b53e4cc0b2a0d5c768866226ddce3de1f57f80c4a02d8d39800fa8e/locks/configure-81e4cd98985eaa00634cd6a4ef41ff86f6ea0a2e1aff632ce10c0f9d6f02e196. Population of work directory failed: Boot binding command failed: exit status: 127

Isolates your Pants from the elements.

Please select from the following boot commands:

<default> (when SCIE_BOOT is not set in the environment)  Detects the current Pants installation and launches it.
bootstrap-tools                                           Introspection tools for the Pants bootstrap process.
update                                                    Update scie-pants.

You can select a boot command by setting the SCIE_BOOT environment variable.
[root@79358947d724 /]#
```

### With this fix:
```
$ docker run --rm -it -v $PWD/dist/scie-pants-linux-x86_64:/usr/bin/pants fedora:37
[root@132281b7c69b /]# pants -V
No Pants configuration was found at or above /.
Would you like to configure / as a Pants project? (Y/n):
Fetching latest stable Pants version since none is configured
Creating /pants.toml and configuring it to use Pants 2.18.2
Bootstrapping Pants 2.18.2
Installing pantsbuild.pants==2.18.2 into a virtual environment at /root/.cache/nce/124c5970822f57509bce2d394c6c2b8384ef32a8c16804c54fcb90dd79f99de1/bindings/venvs/2.18.2
Found existing installation: setuptools 58.1.0
Uninstalling setuptools-58.1.0:
  Successfully uninstalled setuptools-58.1.0
New virtual environment successfully created at /root/.cache/nce/124c5970822f57509bce2d394c6c2b8384ef32a8c16804c54fcb90dd79f99de1/bindings/venvs/2.18.2.
23:42:24.32 [INFO] Initializing scheduler...
23:42:24.50 [INFO] Scheduler initialized.
2.18.2
[root@132281b7c69b /]#
```

File: package/pbt.toml [hunk 1]
```diff
@@ -8,20 +8,20 @@ Python Build Tool: A BusyBox that provides `python`, `pip`, `pex`, `pex3` and `p
 version = "0.7.0"
 
 [lift.scie_jump]
-version = "0.13.0"
+version = "0.14.0"
```
Comment (jsirois) [inline package/pbt.toml:11]: Changes in this helper tool were just drive-by alignment with the scie-pants.toml here and an upgrade below to pick up small size and speed improvements for the scie-pants build process.

File: package/src/main.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ use crate::utils::fs::{base_name, canonicalize, copy, ensure_directory};
 
 const BINARY: &str = "scie-pants";
 
-const SCIENCE_TAG: &str = "v0.2.2";
+const SCIENCE_TAG: &str = "v0.3.1";
```
Comment (jsirois) [inline package/src/main.rs:32]: Another drive-by. This release brings better help when you have lift manifest errors with suggestions for similar names. Release notes:
+ https://github.com/a-scie/lift/blob/main/CHANGES.md#031
+ https://github.com/a-scie/lift/blob/main/CHANGES.md#030

**Comment by jsirois** [comment]
In the past I would release these things, but it's probably best that a current maintainer does that.

**Review by kaos — APPROVED** [review]
Thank you John.
